packages/vitest/src/node/coverage.ts

```diff
@@ -187,7 +192,19 @@ export class BaseCoverageProvider<Options extends ResolvedCoverageOptions<'istan
       includedFiles = (this.ctx.config.related || []).filter(file => includedFiles.includes(file))
     }
 
-    return includedFiles.map(file => slash(path.resolve(this.ctx.config.root, file)))
+    return includedFiles.map(file => slash(path.resolve(root, file)))
+  }
+
+  async getUntestedFiles(testedFiles: string[]): Promise<string[]> {
+    if (this.options.include == null) {
+      return []
+    }
+
+    const rootMapper = this.getUntestedFilesByRoot.bind(this, testedFiles, this.options.include)
```

Comment:
This strategy avoids assigning `this.options.include` to a local variable or ignoring the nullability warning in a lambda.